azkaban-common/src/test/java/azkaban/user/XmlUserManagerTest.java

```diff
@@ -73,6 +79,85 @@ public void testDoNotExist() throws Exception {
     fail("XmlUserManager should throw an exception when the file doesn't exist");
   }
 
+  /**
+   * Test auto reload of user XML
+   */
+  @Test
+  public void testAutoReload() throws Exception {
+    final Props props = new Props(this.baseProps);
+    final URL configURL = Resources.getResource("test-conf/azkaban-users-test1.xml");
+    final String path = configURL.getPath();
+    props.put(XmlUserManager.XML_FILE_PARAM, path);
+
+    UserManager manager;
+    try {
+      manager = new XmlUserManager(props);
+    } catch (final RuntimeException e) {
+      fail("Should have found the xml file");
+      return;
+    }
+
+    // Get the user8 from existing XML with password == password8
+    User user8 = manager.getUser("user8", "password8");
+
+    // Modify the password for user8
+    // TODO : djaiswal : Find a better way to modify XML
+    List<String> lines = new ArrayList<>();
+    List<String> origLines = new ArrayList<>();
+    Path filePath = Paths.get(path);
+    System.out.println("File modification time = " + Files.getLastModifiedTime(filePath).toString());
+    for (String line : Files.readAllLines(filePath)) {
+      origLines.add(line);
+      if (line.contains("password8")) {
+        lines.add(line.replace("password8", "passwordModified"));
+      } else {
+        lines.add(line);
+      }
+    }
+
+    // Make sure the file gets reverted back.
+    try {
+      // Update the file
+      Files.write(filePath, lines);
+      System.out.println("File modification time after write = " + Files.getLastModifiedTime(filePath).toString());
```

Comment:
println here and also in the remainder of the function.

Reply:
It goes to std out, not on the console. It helps in debugging if the test fails.

Reply:
Can this be an assert instead?